src/node/module.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import { default as moduleUtil } from 'node-internal:module';
+import { ERR_INVALID_ARG_VALUE } from 'node-internal:internal_errors';
+
+export function createRequire(path: string | URL): any {
```

Comment:
Could the returned type be refined?

It would help with:

> Note that the return value of createRequire() is a function. In Node.js, this function has additional properties like require.resolve(...), require.main, require.extensions, and require.cache. We are not currently implementing these additional properties but we could in the future if there is a need for them. (require.cache is unlikely to ever be implemented since workers does not have a module cache the way Node.js does).

Reply:
I'm not sure how much compatibility you are looking for but one main purpose of cache is to handle circular dependencies. Nevertheless, we can also implement the missing APIs (or at least a stub of them) in unenv layer if needed, keeping workerd exports for native part.

Reply:
Ah yes, forgot to update that.